### ISSUE TITLE
Allow excluding namespaces

### ISSUE THREAD

**Issue by Zahlii** [open-telemetry/opentelemetry-collector-contrib#23991]
### Component(s)

receiver/k8sevents

### Is your feature request related to a problem? Please describe.

In many scenarios I am not interested in events in. certain namespaces 8e.g. istio, kyma, kube-system). However, I would like to per default monitor all other namespaces, as there are users who can freely create namespaces on the cluster that should automatucally be monitored.

### Describe the solution you'd like

In addition to allow namespaces = [xxx], it would be helpful to also have exclude_namespaces = [xxx], or even better a regex based approach.

**Comment by crobert-1** [comment]
Sounds like a valid request to me.

In the meanwhile, it looks like the namespace is [added as a log attribute](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/k8seventsreceiver/k8s_event_to_logdata.go#L74). Since this is the case, you can use the [filter processor](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/processor/filterprocessor) to drop all logs from the namespaces that you want to exclude. Look at the [alternative config options](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/processor/filterprocessor#alternative-config-options) for logs to see how this could be done.

I'll remove `needs triage` for now, but the code owners may decide to simply rely on the filter processor instead of implementing this in the receiver. I'll defer to their judgement.

**Comment by TylerHelmuth** [comment]
This is a good idea.  Also likely impacted by https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/25161

**Comment by IakimLynnyk-TomTom** [comment]
I don't think this functionality is required in the collector (contrib). This can be configurable with ease. We did this with the [k8sattributes processor](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/processor/k8sattributesprocessor/README.md) and the [filter processor](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/processor/filterprocessor/README.md) in the end.
